### ISSUE TITLE
Using paramatrized configuration in v2 from GitHub action requires dot env files to be available for the action

### ISSUE THREAD

**Issue by itsikbelson-spotlight** [firebase/firebase-functions#1466]
We're migrating to Firebase Functions 2nd gen, and using:
**firebase-functions:** 4.4.1
**firebase-tools:** 12.6.1
**firebase-admin:** 12.6.1
**Node:** 18.15.0
As part of this migration, we're also moving to use[ Parametrized Configuration](https://firebase.google.com/docs/functions/config-env?gen=1st#params) instead of Runtime Config, which is no longer supported in 2nd gen functions.

I may be totally off here, but want to get some feedback.

We're using GitHub actions and GitHub runners to deploy our Firebase Functions to the different environment.
My understanding is that with the new setup of configuration, during deployment the `firebase deploy` command takes the settings from dot env files and saves them as the Cloud Function variables.
However, since the deployment runs on a GitHub runner, it needs access to these dot env files, which implies the dot env files need to be part of the source code.
However, saving dot env files in source code is considered a bad practice.
I know that for secrets we can use secret manager, and in this case the dot env files do not include any secrets.
Still, it seems odd to me to put env files in source code.

Are there any best practices for handling that?
Would love to hear what others who uses similar setup do.

Thanks !
Itsik

**Comment by blidd-google** [comment]
Hi @itsikbelson-spotlight, as you said, parameterized configuration in 2nd gen will attempt to load the values from local .env files; this should be secure if you parameterize secrets using the `Secret` parameter type (which will store and load those parameterized values in Secret Manager). Parameterizing your environment configuration also provide additional benefits such as strongly typed variables and validation at deploy time. Our view is that such a setup is safe for CI/CD workflows in GitHub Actions (again, as long as secrets are parameterized using the Secret type). As this is intended behavior I'm gonna go ahead and close the issue, but others can still chime in on how they've set up their CI/CD pipelines using parameterized configuration.

**Comment by itsikbelson-spotlight** [comment]
Thanks @blidd-google 
I'm still missing a way to set these parameters from the outside, something to replace `firebase functions:config:set`...

**Comment by boyfunky** [comment]
@itsikbelson-spotlight did you find a way to resolve this? I am also facing the same issues

**Comment by itsikbelson-spotlight** [comment]
> @itsikbelson-spotlight did you find a way to resolve this? I am also facing the same issues

Not really, but I haven't looked it there's something new in Firebase for this for a while.
Our future plans (if Firebase does not provide another way to do it) is to store the env config somewhere else (e.g. cloud storage), and then use a secret key to access this config (in Cloud Storage).
One more thing which will probably have to remain in the Firebase functions parameterized configuration is the params required for the function deployment (e.g. min/max instances)
